src/libexpr/value.hh

```diff
@@ -286,34 +288,25 @@ public:
             abort();
     }
 
-    /**
-     * After overwriting an app node, be sure to clear pointers in the
-     * Value to ensure that the target isn't kept alive unnecessarily.
-     */
-    inline void clearValue()
+    inline void finishValue(InternalType newType, Payload newPayload)
```

Comment:
Should it be like the following, or does the inlining take care of unnecessary copying? I would think that `newPayload` must still be the equivalent of a normal local variable after inlining.

```suggestion
    inline void finishValue(InternalType newType, Payload && newPayload)
```